### ISSUE TITLE
Error Report

### ISSUE THREAD

**Issue by kittenboops13** [magefree/mage#7239]
uhhh I think this is what I'm supposed to do

```
Game exception occurred: mage.MageException: Iterated player priority after game exception too often, game ends!
Server version: 1.4.47V0-beta3 (build: 2020-11-30 02:35)
mage.game.GameImpl.playPriority(GameImpl.java:1397)
mage.game.turn.Step.priority(Step.java:61)
mage.game.turn.CombatDamageStep.priority(CombatDamageStep.java:31)
mage.game.turn.Phase.playStep(Phase.java:184)
mage.game.turn.Phase.play(Phase.java:89)
mage.game.turn.Turn.play(Turn.java:126)
mage.game.GameImpl.playTurn(GameImpl.java:893)
mage.game.GameImpl.play(GameImpl.java:805)
mage.game.GameImpl.start(GameImpl.java:790)
mage.server.game.GameWorker.call(GameWorker.java:34)
java.util.concurrent.FutureTask.run(FutureTask.java:266)
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
java.lang.Thread.run(Thread.java:748)
```

**Comment by theelk801** [comment]
Can you give some details on what was happening during the game?

**Comment by JayDi85** [comment]
Real error:
```
FATAL 2020-12-11 23:18:56,962 Game exception gameId: 4a15713e-03f8-43f9-8d5c-56f7c58c5137                                =>[GAME 4a15713e-03f8-43f9-8d5c-56f7c58c5137] GameImpl.playPriority
java.lang.ClassCastException: java.util.ArrayList cannot be cast to mage.players.PlayerList
        at mage.watchers.common.PlayersAttackedThisTurnWatcher.watch(PlayersAttackedThisTurnWatcher.java:36)
        at mage.watchers.Watchers.watch(Watchers.java:34)
        at mage.game.GameState.handleEvent(GameState.java:759)
        at mage.game.GameState.handleSimultaneousEvent(GameState.java:734)
        at mage.game.GameImpl.checkStateAndTriggered(GameImpl.java:1808)
        at mage.game.GameImpl.playPriority(GameImpl.java:1343)
        at mage.game.turn.Step.priority(Step.java:61)
        at mage.game.turn.Phase.playStep(Phase.java:184)
        at mage.game.turn.Phase.play(Phase.java:89)
        at mage.game.turn.Turn.playExtraPhases(Turn.java:231)
        at mage.game.turn.Turn.play(Turn.java:135)
        at mage.game.GameImpl.playTurn(GameImpl.java:893)
        at mage.game.GameImpl.play(GameImpl.java:805)
        at mage.game.GameImpl.start(GameImpl.java:790)
        at mage.server.game.GameWorker.call(GameWorker.java:34)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

**Comment by kittenboops13** [comment]
Ah, yes, I suppose that might be helpful
I was playing Commander one-on-one against the AI. It was swinging some Rat tokens at me, which I went to block with Gleaming Barrier. That's when I got the error.

**Comment by jeffwadsworth** [comment]
What other cards did you have in play?  Something cared about "Players that attacked this turn" and that wasn't Gleaming Barrier.

**Comment by kittenboops13** [comment]
I don't remember which other cards I had in play, as this was a couple days ago now, but I managed to get the same exact error playing the same two decks against each other just now. I had out both my commanders, Vial Smasher and Malcolm, Keen-Eyed, Dire Fleet Hoarder, Gleaming Barrier (again, funny enough), a Mind Stone, and some basic lands.
The only cards the AI had out, aside from basic lands, were the commander Marrow-Gnawer and an Altar of Dementia.
I had just selected "All Attack" and the AI went to block my Dire Fleet Hoarder. That's when the exact same error message popped up again.
